src/bos/server/controllers/v2/boot_set.py

```diff
@@ -171,6 +203,36 @@ def _validate_boot_set(bs: dict, operation: str, reject_nids: bool) -> list[str]
 
     return warning_msgs
 
+
+def validate_boot_set_arch(bs: dict, image_metadata: BootImageMetaData|None=None) -> None:
+    """
+    If the boot set architecture is not set to Other, check that the IMS image
+    architecture matches the boot set architecture (treating a boot set architecture
+    of Unknown as X86)
+    """
+    arch = bs.get("arch", DEFAULT_ARCH)
+    if arch == 'Other':
+        raise CannotValidateBootSetArch("Boot set arch set to 'Other'")
+
+    if image_metadata is None:
+        try:
+            image_metadata = BootImageMetaDataFactory(bs)()
+        except Exception as err:
+            raise CannotValidateBootSetArch(
+                f"Can't find boot artifacts: {exc_type_msg(err)}") from err
+
+    try:
+        ims_image_arch = image_metadata.arch
+    except Exception as err:
```

Comment:
Are you expecting the `arch` attribute to not exist in this case?

Reply:
Actually, I think as written, this `except` block will not be used. This is leftover from when I had some fatal errors being raised inside the `arch` method. But then we decided to make those only fatal if certain options are set, and I changed the `arch` method accordingly, but didn't change this part. I can take it out, and it can be added back when I implement the aforementioned options.